### PR TITLE
Update documentation for MIG

### LINKED ISSUE #146: Add a new section about MIG in the node profile description
Briefly explain the MIG feature and partitioning used on the Mila cluster in a new section at https://docs.mila.quebec/Information.html#node-profile-description.

https://www.nvidia.com/en-us/technologies/multi-instance-gpu/

### PR DESCRIPTION
**Pull request by obilaniu**
Describe new NVIDIA MIG GPU feature and SLURM syntaxes allowed. Warn about limitations.

Closes #146 .

### CODE REVIEW

**Review by satyaog — APPROVED** [review]

File: docs/Information_nodes.rst [hunk 1]
```diff
@@ -70,6 +70,60 @@ commands::
...
+| ``a100l.2g.20gb``:h:`<p>`   |          | 20GB :h:`<p>` | 2/7th       |     108      |
+| ``a100l.2``                 | A100     | (2/8th)       | *of full*   |              |
++-----------------------------+----------+---------------+-------------+--------------+
+| ``a100l.3g.40gb``:h:`<p>`   |          | 40GB :h:`<p>` | 3/7th       |     72       |
+| ``a100l.3``                 | A100     | (4/8th)       | *of full*   |              |
++-----------------------------+----------+---------------+-------------+--------------+
+
+And can be requested using a SLURM flag such as ``--gres=gpu:a100l.1``
+
+The partitioning may be revisted as needs and SLURM capabilities evolve. Other
```
Comment (btravouillon) [inline docs/Information_nodes.rst:111]: revised? or revisited?

Comment (obilaniu) [reply docs/Information_nodes.rst:111]: In my head I had "revisited" but with my typos it's exactly Levenshtein edit distance=1 from two alternatives that have the same meaning here. Which would you like?

Comment (btravouillon) [reply docs/Information_nodes.rst:111]: One of both. :stuck_out_tongue_closed_eyes:

Comment (obilaniu) [reply docs/Information_nodes.rst]: Revised then. 2 bytes and 1 syllable shorter than the alternative.

File: docs/Information_nodes.rst [hunk 2]
```diff
@@ -70,6 +70,60 @@ commands::
...
+^^^
+
+.. _mig_nodes:
+
+MIG (`Multi-Instance GPU <https://www.nvidia.com/en-us/technologies/multi-instance-gpu/>`_)
+is an NVIDIA technology allowing certain GPUs to be
+partitioned into multiple *instances*, each of which has a roughly proportional
+amount of compute resources, device memory and bandwidth to that memory.
+
+NVIDIA supports MIG on its A100 GPUs and allows slicing the A100 into up to 7
```
Comment (fosterrath-mila) [inline docs/Information_nodes.rst:83]: theoretically allows ?

**Review by fosterrath-mila — APPROVED** [review]

File: docs/Information_nodes.rst (hunk 2 above)
Comment (satyaog) [reply docs/Information_nodes.rst:83]: I think it's ok as it is but I'll let @obilaniu decide. However, if tomorrow same time this is not merged I'll press the button ;)

Comment (obilaniu) [reply docs/Information_nodes.rst:83]: The "theoretically" discussion serves to introduce the explanation that the static-partitioning issue isn't NVIDIA's but SLURM's fault, and thus that this partitioning isn't set in stone and is changeable (and we likely will change it). I'd like to keep it but if it looks too much like going into the weeds, we can take it out.

File: docs/Information_nodes.rst [hunk 3]
```diff
@@ -70,6 +70,60 @@ commands::
...
+instances. Although this can theoretically be done dynamically, the SLURM job
+scheduler does not support doing so in practice as it does not model
+reconfigurable resources very well. Therefore, the A100s must currently be
+statically partitioned into the required number of instances of every size
+expected to be used.
+
+The new ``cn-g`` series of nodes include A100-80GB GPUs. One third have been
```
Comment (btravouillon) [inline docs/Information_nodes.rst:90]:
```suggestion
The ``cn-g`` series of nodes include A100-80GB GPUs. One third have been
```

I would remove the temporality here. Just make the documentation accurate for the years to come. ;)

File: docs/Information_nodes.rst [hunk 4]
```diff
@@ -70,6 +70,60 @@ commands::
...
+
+And can be requested using a SLURM flag such as ``--gres=gpu:a100l.1``
+
+The partitioning may be revised as needs and SLURM capabilities evolve. Other
+MIG profiles exist and could be introduced.
+
+
+.. warning::
+
+    MIG has a number of `severe limitations <https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html#app-considerations>`_,
```
Comment (btravouillon) [inline docs/Information_nodes.rst:117]:
```suggestion
    MIG has a number of `important limitations <https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html#app-considerations>`_,
```
I would even rephrase it to make it less dramatic. I mean, we chose to enable MIG on 2/3 of our A100 80GB GPUs, that should not be that severe.

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍
